src/V3Timing.cpp

```diff
@@ -0,0 +1,657 @@
+        AstVarScope* forkVscp
+            = createTemp(flp, forkp->name() + "__sync", getCreateForkSyncDTypep(), insertBeforep);
+        unsigned joinCount = 0;  // Needed for join counter
+        // Add a <fork sync>.done() to each begin
+        for (AstNode* beginp = forkp->stmtsp(); beginp; beginp = beginp->nextp()) {
+            addForkDone(VN_AS(beginp, Begin), forkVscp);
+            joinCount++;
+        }
+        if (forkp->joinType().joinAny()) joinCount = 1;
+        // Set the join counter
+        auto* const initp = new AstCMethodHard{flp, new AstVarRef{flp, forkVscp, VAccess::WRITE},
+                                               "init", new AstConst{flp, joinCount}};
+        initp->dtypeSetVoid();
+        initp->statement(true);
+        forkp->addHereThisAsNext(initp);
+        // Await the join at the end
+        auto* const joinp
+            = new AstCMethodHard{flp, new AstVarRef{flp, forkVscp, VAccess::WRITE}, "join"};
+        joinp->dtypeSetVoid();
+        // Add debug info
+        joinp->addPinsp(new AstText{flp, '"' + flp->filename() + '"'});
+        joinp->addPinsp(new AstText{flp, cvtToStr(flp->lineno())});
+        auto* const awaitp = new AstCAwait{flp, joinp};
+        awaitp->statement(true);
+        forkp->addNextHere(awaitp);
+    }
+
+    // VISITORS
+    virtual void visit(AstNodeModule* nodep) override {
+        UASSERT(!m_classp, "Module or class under class");
+        VL_RESTORER(m_classp);
+        m_classp = VN_CAST(nodep, Class);
+        VL_RESTORER(m_timescaleFactor);
+        m_timescaleFactor = calculateTimescaleFactor(nodep->timeunit());
+        iterateChildren(nodep);
+    }
+    virtual void visit(AstScope* nodep) override {
+        VL_RESTORER(m_scopep);
+        m_scopep = nodep;
+        iterateChildren(nodep);
+    }
+    virtual void visit(AstActive* nodep) override {
+        m_activep = nodep;
+        iterateChildren(nodep);
+        m_activep = nullptr;
+    }
+    virtual void visit(AstNodeProcedure* nodep) override {
+        VL_RESTORER(m_procp);
+        m_procp = nodep;
+        iterateChildren(nodep);
+        if (nodep->user2()) nodep->setSuspendable();
+    }
+    virtual void visit(AstAlways* nodep) override {
+        visit(static_cast<AstNodeProcedure*>(nodep));
+        if (nodep->isSuspendable() && !nodep->user1SetOnce()) {
+            FileLine* const flp = nodep->fileline();
+            AstSenTree* const sensesp = m_activep->sensesp();
+            if (sensesp->hasClocked()) {
+                AstNode* bodysp = nodep->bodysp()->unlinkFrBackWithNext();
+                auto* const controlp = new AstEventControl{flp, sensesp->cloneTree(false), bodysp};
+                nodep->addStmtp(controlp);
+                iterate(controlp);
+            }
+            auto* const activep = new AstActive{
+                flp, "", new AstSenTree{flp, new AstSenItem{flp, AstSenItem::Initial{}}}};
```

Comment:
```suggestion
            // Note: The 'while (true)' outer loop will be added in V3Sched
            auto* const activep = new AstActive{
                flp, "", new AstSenTree{flp, new AstSenItem{flp, AstSenItem::Initial{}}}};
```

At some point I might add a different SenItem for this 'initial always', purely for cosmetic reasons (to help profilers), at which point the while should be added here, but I'm happy as it is otherwise.